operators/slang_shader/slang_shader_op.cpp

```diff
@@ -0,0 +1,344 @@
+              "Attribute '{}' supports scalar uniforms only and cannot be applied to '{}'.",
+              attrib_name,
+              parameter["name"].get<std::string>()));
+        }
+        const std::string param_name = user_attrib["arguments"].at(0);
+        if (parameter["type"]["scalarType"] == "bool") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<bool>(), param_name, parameter["binding"]["offset"]));
+        } else if (parameter["type"]["scalarType"] == "int32") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<int32_t>(), param_name, parameter["binding"]["offset"]));
+        } else if (parameter["type"]["scalarType"] == "uint32") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<uint32_t>(), param_name, parameter["binding"]["offset"]));
+        } else if (parameter["type"]["scalarType"] == "int64") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<int64_t>(), param_name, parameter["binding"]["offset"]));
+        } else if (parameter["type"]["scalarType"] == "uint64") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<uint64_t>(), param_name, parameter["binding"]["offset"]));
+        } else if (parameter["type"]["scalarType"] == "float32") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<float>(), param_name, parameter["binding"]["offset"]));
+        } else if (parameter["type"]["scalarType"] == "float64") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<double>(), param_name, parameter["binding"]["offset"]));
+        } else if (parameter["type"]["scalarType"] == "int8") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<int8_t>(), param_name, parameter["binding"]["offset"]));
+        } else if (parameter["type"]["scalarType"] == "uint8") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<uint8_t>(), param_name, parameter["binding"]["offset"]));
+        } else if (parameter["type"]["scalarType"] == "int16") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<int16_t>(), param_name, parameter["binding"]["offset"]));
+        } else if (parameter["type"]["scalarType"] == "uint16") {
+          impl_->pre_launch_commands_.push_back(std::make_unique<CommandParameter>(
+              spec, new Parameter<uint16_t>(), param_name, parameter["binding"]["offset"]));
+        } else {
+          throw std::runtime_error(
```

Comment:
a helper function may make this look cleaner.

Reply:
Good suggestion, done.